packages/admin/admin/src/form/Field.tsx

```diff
@@ -62,12 +62,26 @@ export function Field<FieldValue = any, FieldElement extends HTMLElement = HTMLE
     const shouldShowWarning = passedShouldShowWarning ?? finalFormContext.shouldShowFieldWarning;
     const shouldScrollToField = passedShouldScrollTo ?? finalFormContext.shouldScrollToField;
 
+    // We need to change the key of the field when required or validate changes, to force revalidation
+    // https://codesandbox.io/p/sandbox/changing-field-level-validators-zc8ei
+    const keyRequired = required ? "required" : "not-required";
+    const keyValidate = validate ? "validate" : "no-validate";
```

Comment:
alternatively you could implement the `validateError` function in a way that it doesn't get re-created based on required.

naive implementation:
```
const validateError = (value: any, allValues: object) => {
    if (required) {
        const requiredReturn = requiredValidator(value, allValues);
        if (requiredReturn !== undefined) return requiredReturn;
    }
    if (validate) {
        return validate(value, allValues);
    }
    return undefined
};
```

Reply:
To undersand the issue. see note in docs: https://final-form.org/docs/react-final-form/types/FieldProps#validate (and the codesandbox you linked is also from there)

Reply:
@nsams in your naive implementation, required would probably be a stale value